macro/src/lib.rs

```diff
@@ -32,18 +32,78 @@ fn param_ty(param: &ast::Param) -> syn::Type {
             // not Rust stdlib types (which are not FFI-safe and must be converted)
             prim.get_diplomat_slice_type(ltmt)
         }
-        _ => param.ty.to_syn(),
+        _ => param_ty.to_syn(),
     }
 }
 
-fn param_conversion(param: &ast::Param) -> Option<proc_macro2::TokenStream> {
-    let name = &param.name;
-    match &param.ty {
+fn orig_syn(ty: &ast::TypeName) -> syn::Type {
```

Comment:
issue: as discussed in chat, this isn't necessary, `ast::TypeName.to_syn()` is expected to roundtrip, and if it doesn't, that's a bug (we identified the bug in the PrimitiveSlice's to_syn impl)

the macro uses `to_syn()` to get the user-specified type, and `param_ty` to convert it to the ffi-safe type